packages/sample-controllers/src/sample-gas-prices-service/sample-gas-prices-service.ts

```diff
@@ -11,46 +75,91 @@ type GasPricesResponse = {
+   */
+  messenger: SampleGasPricesServiceMessenger;
+  /**
+   * A function that can be used to make an HTTP request. If your JavaScript
+   * environment supports `fetch` natively, you'll probably want to pass that;
+   * otherwise you can pass an equivalent (such as `fetch` via `node-fetch`).
+   */
+  fetch: typeof fetch;
+  /**
+   * Options to pass to `createServicePolicy`, which is used to wrap each
+   * request.
+   *
+   * @see {@link CreateServicePolicyOptions}
+   */
+  policyOptions?: CreateServicePolicyOptions;
+};
+
 /**
  * This service object is responsible for fetching gas prices via an API.
  *
  * @example
  *
- * On its own:
- *
  * ``` ts
- * const gasPricesService = new SampleGasPricesService({ fetch });
- * // Fetch gas prices for Mainnet
- * const gasPricesResponse = await gasPricesService.fetchGasPrices('0x1');
- * // ... Do something with the response ...
- * ```
+ * import type { SampleGasPricesMessenger } from '@metamask/sample-controllers';
  *
- * In conjunction with `SampleGasPricesController`:
+ * const messenger: SampleGasPricesMessenger = new Messenger();
+ * const service = new SampleGasPricesService({ messenger, fetch });
  *
- * ``` ts
- * const gasPricesService = new SampleGasPricesService({ fetch });
- * const gasPricesController = new SampleGasPricesController({
- *   // ... state, messenger, etc. ...
- *   gasPricesService,
- * });
- * // This will use the service object internally
- * gasPricesController.updateGasPrices();
+ * // Fetch gas prices for Mainnet
+ * const gasPricesResponse = await service.fetchGasPrices('0x1');
+ *
+ * // ... Do something with the response ...
  * ```
  */
 export class SampleGasPricesService {
-  readonly #fetch: typeof fetch;
+  /**
+   * The messenger suited for this service.
+   */
+  readonly #messenger: SampleGasPricesServiceOptions['messenger'];
+
+  /**
+   * A function that can be used to make an HTTP request.
+   */
+  readonly #fetch: SampleGasPricesServiceOptions['fetch'];
+
+  /**
+   * The policy that wraps the request.
+   *
+   * @see {@link createServicePolicy}
+   */
+  readonly #policy: ServicePolicy;
 
   /**
    * Constructs a new SampleGasPricesService object.
    *
-   * @param args - The arguments.
-   * @param args.fetch - A function that can be used to make an HTTP request.
-   * If your JavaScript environment supports `fetch` natively, you'll probably
-   * want to pass that; otherwise you can pass an equivalent (such as `fetch`
-   * via `node-fetch`).
+   * @param options - The options. See {@link SampleGasPricesServiceOptions}.
    */
-  constructor({ fetch: fetchFunction }: { fetch: typeof fetch }) {
+  constructor(options: SampleGasPricesServiceOptions) {
+    const { messenger, fetch: fetchFunction, policyOptions = {} } = options;
+
+    this.#messenger = messenger;
     this.#fetch = fetchFunction;
+    this.#policy = createServicePolicy(policyOptions);
+
+    this.#messenger.registerActionHandler(
```

Comment:
Nit: should we update these examples to use the new `registerMethodActionHandlers` method?

Reply:
Yes, we might as well. Other teams are starting to hand-write similar patterns. Done here: [7e967d2](https://github.com/MetaMask/core/pull/6168/commits/7e967d20853aa613f6799d91e8929db7db7d1114)